homeassistant/helpers/entity_component.py

```diff
@@ -131,14 +131,18 @@ def _async_setup_platform(self, platform_type, platform_config,
         scan_interval = (platform_config.get(CONF_SCAN_INTERVAL) or
                          getattr(platform, 'SCAN_INTERVAL', None) or
                          self.scan_interval)
+        scan_sequential = getattr(platform, 'PARALLEL_UPDATES', 1)
```

Comment:
So, this is just an example, but I'm wondering if we should do something like this:
```python
if hasattr(platform, 'PARALLEL_UPDATES'):
    scan_sequential = getattr(platform, 'PARALLEL_UPDATES')
else:
    # async platforms default to unlimited parallel updates. sync platforms 
    # default to half available workers as a safeguard to ensure the whole
    # thread pool isn't accidentally locked.
    scan_sequential = 0 if hasattr(platform, 'async_setup_platform') else pool.max_workers / 2
```

Reply:
We can change our handling deeps on if it is a async platform or not but if not we should set it to 1 like it is now. The part of pool size make no sense.

Reply:
I think that we should allow people to do as much in parallel as they want until we see platforms that will violate this.

Let's do best of both:

```
scan_sequential = getattr(platform, 'PARALLEL_UPDATES', int(hasattr(platform, 'async_setup_platform')))
```